src/Cardano/Transaction/Balancer/Balancer.purs

```diff
@@ -189,18 +203,24 @@ runBalancer unbalancedTx ctx = do
         ctx.pparams
         changeAddress
         txWithNetwork
+        spendableUtxos
+  let txWithMinAda = addLovelacesToTransactionOutputs ctx.pparams txWithCollateral
+  -- Get collateral inputs to mark them as unspendable.
+  -- Some CIP-30 wallets don't allow to sign Txs that spend them.
```

Comment:
Hmm, this may cause UX issues, like e.g. a wallet marks too much ADA as collateral and even when a user has enough ADA for a transaction, it fails telling him not enough ADA...

Reply:
I see your point, but I think the current approach is more stable, assuming that some wallets may restrict spending of collateral utxos